pkg/pillar/devicenetwork/dnc.go

```diff
@@ -796,6 +797,26 @@ func IngestPortConfigList(ctx *DeviceNetworkContext) {
 				portConfig.Key)
 			continue
 		}
+
+		var invalidCert bool
+		caCertPool := x509.NewCertPool()
+		for _, port := range portConfig.Ports {
+			for _, pem := range port.ProxyCertPEM {
+				if !caCertPool.AppendCertsFromPEM(pem) {
+					invalidCert = true
+					break
+				}
+			}
+			if invalidCert {
+				break
+			}
+		}
+		if invalidCert {
+			log.Warnf("Stored DevicePortConfig key %s contains invalid certificate; ignored",
```

Comment:
Just out of curiosity, did we see bad PEM content in the test which caused the crash? I see you founnd and fixed code which prints the whole PEM content if it is bad.

Reply:
A customer submitted a badly formatted certificate. I have copied it from their logs and use it for testing purposes.
So nothing wrong in our tests.